src/instructlab/model/list.py

```diff
@@ -0,0 +1,138 @@
+    return f_size, magnitude
+
+
+# print_table displays the models on the system in a nicely formatted table
+def print_table(headers: list[str], data: list[list[str]]):
+    column_widths = [
+        max(len(str(row[i])) for row in data + [headers]) for i in range(len(headers))
+    ]
+    for i, header in enumerate(headers):
+        print(f"{header:{column_widths[i]}}", end=" | ")
+    print()
+
+    # Print separator line between headers and data
+    print("-" * (sum(column_widths) + 3 * len(headers) - 1))
+
+    for row in data:
+        for i, item in enumerate(row):
+            print(f"{item:{column_widths[i]}}", end=" | ")
+        print()
+
+
+AnalyzeResultGGUF = list[str]
+AnalyzeResultDir = list[list[str]]
+
+
+def _analyze_gguf(entry: Path) -> AnalyzeResultGGUF:
+    # stat the gguf, add it to the table
+    stat = Path(entry.absolute()).stat(follow_symlinks=False)
+    f_size = stat.st_size
+    f_size, magnitude = convert_bytes_to_proper_mag(f_size)
+    # add to table
+    age = pandas.to_timedelta(time.time_ns() - stat.st_mtime_ns)
+    return [entry.name, str(age), f"{f_size} {magnitude}"]
+
+
+def _analyze_dir(
+    entry: Path, list_checkpoints: bool, directory: str
+) -> AnalyzeResultDir:
+    actual_model_name = ""
+    all_files_sizes = 0
+    add_model = False
+    models = []
+    # walk entire dir.
+    for root, _, files in os.walk(entry.as_posix()):
+        normalized_path = os.path.normpath(root)
+        # Split the path into its components
+        parts = normalized_path.split(os.sep)
+        # Get the last two parts and join them back into a path
+        last_two_parts = os.path.join(parts[-2], parts[-1])
+        # if this is a dir it could be:
+        # top level repo dir `instructlab/`
+        # top level model dir `instructlab/granite-7b-lab`
+        # checkpoint top level dir `step-19`
+        # any lower level dir: `instructlab/granite-7b-lab/.huggingface/download.....`
+        # so, check if model is valid Safetensor, GGUF, or list it regardless w/ `--list-checkpoints`
+        # if --list-checkpoints is specified, we will list all checkpoints in the checkpoints dir regardless of the ModelCard
+        if is_model_safetensors(Path(normalized_path)) or is_model_gguf(
+            Path(normalized_path)
+        ):
+            actual_model_name = last_two_parts
+            all_files_sizes = 0
+            add_model = True
+        else:
+            if list_checkpoints and directory is DEFAULTS.CHECKPOINTS_DIR:
+                logging.debug("Including model regardless of model validity")
+            else:
+                continue
+        for f in files:
+            # stat each file in the dir, add the size in Bytes, then convert to proper magnitude
+            full_file = os.path.join(root, f)
+            stat = Path(full_file).stat(follow_symlinks=False)
+            all_files_sizes += stat.st_size
+        all_files_sizes, magnitude = convert_bytes_to_proper_mag(all_files_sizes)
```

Comment:
(nit) this seems unused if `add_model` is `False`. Should this move under the `if`? (Why do we even run the code above if it's not `add_model`? Seems like a waste of CPU? What am I missing?